Dom-mainpulation.js

```diff
@@ -496,4 +496,78 @@ function addModal4() {
 }
 
 document.getElementById('see-project4').addEventListener('click', addModal4);
-document.getElementById('see-project-d4').addEventListener('click', addModal4);
\ No newline at end of file
+document.getElementById('see-project-d4').addEventListener('click', addModal4);
+
+// Form validation
+const form = document.getElementById('form');
+const User = document.getElementById('name');
+const email = document.getElementById('email');
+const textarea = document.getElementById('textarea');
+const errorBtn = document.querySelector('.error-div-btn');
+const FormError = document.querySelector('.form-control-btn');
+
+function setErrorForBtn(input, message) {
+  const formControl = input.parentElement;
+  const small = formControl.querySelector('small');
+  small.innerText = message;
+  formControl.className = 'form-control-btn error';
+}
+
+function setErrorFor(input, message) {
+  const formControl = input.parentElement;
+  const small = formControl.querySelector('small');
+  small.innerText = message;
+  formControl.className = 'form-control error';
+}
+
+function setSuccessFor(input) {
+  const formControl = input.parentElement;
+  formControl.className = 'form-control success';
+}
+
+function validateImputs() {
+  const nameValue = User.value.trim();
+  const emailValue = email.value.trim();
+  const textValue = textarea.value.trim();
+
+  if (nameValue === '') {
+    setErrorFor(User, 'Name cannot be blank');
+    setErrorForBtn(errorBtn, 'All inputs must be valid');
+  } else if (nameValue.length >= 30) {
+    setErrorFor(User, 'Name must be less than 30 characters long');
+    setErrorForBtn(errorBtn, 'All inputs must be valid');
+  } else {
+    setSuccessFor(User);
+    setSuccessFor(errorBtn);
+  }
+
+  if (emailValue === '') {
+    setErrorFor(email, 'Email cannot be blank');
+    setErrorForBtn(errorBtn, 'All inputs must be valid');
+  } else if ((emailValue === String(emailValue).toLowerCase()) === false) {
+    setErrorFor(email, 'Email must be in lowercase');
+    setErrorForBtn(errorBtn, 'All inputs must be valid');
+  } else {
+    setSuccessFor(email);
+    setSuccessFor(errorBtn);
+  }
+
+  if (textValue === '') {
+    setErrorFor(textarea, 'Message cannot be empty');
+    setErrorForBtn(errorBtn, 'All inputs must be valid');
+  } else if (textValue.length >= 500) {
+    setErrorFor(textarea, 'Message must be under 500 characters long');
+    setErrorForBtn(errorBtn, 'All inputs must be valid');
+  } else {
+    setSuccessFor(textarea);
+    setSuccessFor(errorBtn);
+  }
+}
+
+form.addEventListener('submit', (e) => {
+  validateImputs();
+
+  if (FormError.className === 'form-control-btn error') {
+    e.preventDefault();
+  }
+});
```

Comment:
- [ ] Great job so far but As I can see if I write my email in capital case it shows me the error that's good but also it submit my message but according to the requirements you shouldn't let the form submit if there is capital case in the email it should show an error message and then the user can fix the email so kindly work on that and fix it 🚀💪🏻💯🔥


https://user-images.githubusercontent.com/58553711/195734673-f71827a7-1b9b-4fd8-8a04-caeb3aef4cdc.mp4